src/cpl_nuopc/wav_comp_nuopc.F90

```diff
@@ -1482,24 +1216,175 @@ subroutine ModelFinalize(gcomp, rc)
     character(len=*),parameter  :: subname=trim(modName)//':(ModelFinalize) '
     !-------------------------------------------------------------------------------
 
-    !--------------------------------
-    ! Finalize routine
-    !--------------------------------
-
-   print*, 'HK model finalize: ',  dtmax, dtcfl, dtcfli, dtmin
     rc = ESMF_SUCCESS
-    call ESMF_LogWrite(subname//' called', ESMF_LOGMSG_INFO)
+    call ESMF_LogWrite(trim(subname)//' called', ESMF_LOGMSG_INFO)
 
-    if (masterproc) then
+    if (root_task) then
        write(stdout,F91)
        write(stdout,F00) 'WW3: end of main integration loop'
        write(stdout,F91)
     end if
 
-    call ESMF_LogWrite(subname//' done', ESMF_LOGMSG_INFO)
+    call ESMF_LogWrite(trim(subname)//' done', ESMF_LOGMSG_INFO)
 
   end subroutine ModelFinalize
 
-  !===============================================================================
+!===============================================================================
+#ifdef CESMCOUPLED
+  subroutine waveinit_cesm(gcomp, ntrace, mpi_comm, dtime_sync, rc)
+
+    ! Initialize ww3 for cesm (called from InitializeRealize)
+
+    use w3gdatmd     , only : dtcfl, dtcfli, dtmax, dtmin
+    use w3initmd     , only : w3init
+    use w3odatmd     , only : nds
+    use w3initmd     , only : w3init
+    use wav_shr_mod  , only : casename, initfile, outfreq
+    use wav_shr_mod  , only : inst_index, inst_name, inst_suffix
+    use wav_shel_inp , only : set_shel_inp
+    use wav_shel_inp , only : npts, odat, iprt, x, y, pnames, prtfrm
+    use wav_shel_inp , only : flgrd, flgd, flgr2, flg2
+
+    ! input/output variables
+    type(ESMF_GridComp)   :: gcomp
+    integer , intent(in)  :: ntrace(:)
+    integer , intent(in)  :: mpi_comm
+    integer , intent(in)  :: dtime_sync
+    integer , intent(out) :: rc
+
+    ! local variables
+    integer                        :: ierr
+    integer                        :: unitn  ! namelist unit number
+    integer                        :: shrlogunit
+    logical                        :: isPresent, isSet
+    character(len=CL)              :: cvalue
+    character(len=*), parameter    :: subname = '(wav_comp_nuopc:wavinit_cesm)'
+    ! -------------------------------------------------------------------
+
+    rc = ESMF_SUCCESS
+    if (dbug_flag > 5) call ESMF_LogWrite(trim(subname)//' called', ESMF_LOGMSG_INFO)
+
+    namelist /ww3_inparm/ initfile, outfreq, dtcfl, dtcfli, dtmax, dtmin
```

Comment:
This namelist declaration should be moved above the executable statements in lines 1264-1265.

Reply:
Done.